ls1intum/Artemis
# `Assessment`: Fix type error in the course score calculation

### Checklist
#### General
#### Client

### Motivation and Context
Closes [ARTEMIS-M27](https://sentry.ase.in.tum.de/organizations/artemis/issues/20984/events/e925aa5ff84c4c3680cc2026832a1a54/?project=2)

### Description
`buildAndTestStudentSubmissionsAfterDueDate` is not always correctly converted to a dayjs object. This is because calls to general exercise-apis only covert standard dates into dayjs()-objects, not dates specific to one exercise type. By changing the call order this conversion is done implicitly in the isAfter()-call.

I also changed a comparison that for certain was a small overlook. 

### Steps for Testing
Code Review is most important. You can:
1. Open the Artemis front page and the course statistic page
2. Make sure no errors are present in the console.

### Review Progress
#### Code Review
#### Manual Tests
